explainer.md

```diff
@@ -1,751 +1,423 @@
-  
-  // The start() call triggers an async operation to capture
-  // snapshots for the offered elements,
-  await transition.start(async () => {
-    // This callback is invoked by the browser when the capture
-    // finishes and the DOM can be switched to the new state.
-    // No frames are rendered until this callback returns.
-    
-    // Asynchronously load the new page.
-    await coolFramework.changeTheDOMToPageB();
-    
-    // Clear the old message if that element is still in the page
-    document.querySelector(".old-message").style.pageTransitionTag = "";
-    // Set new message as the shared element 'message'
-    document.querySelector(".new-message").style.pageTransitionTag = "message";
-    
-    // Set up animations using WA-API on the next frame.
-    requestAnimationFrame(() => {
-      document.documentElement.animate(keyframes, {
-      ...animationOptions,
-      pseudoElement: "::page-transition-container(message)",
-      });
-    });
-    
-    // Note that when this callback finishes, the animations will start with the tagged elements.
-  });
+The rough plan is to allow nesting via an opt-in (all API names used here are for entertainment purposes only):
+
+```css
+.container {
+  page-transition-tag: container;
+  contain: paint;
+}
+.child-item {
+  page-transition-tag: child-item;
+  contain: layout;
+  page-transition-style-or-whatever: nested;
 }
 ```
 
-- `transition.start(callback)` - Asynchornously capture any offered element as
-  "outgoing". When finished, invoke the callback. When the callback finishes
-  running, capture any new elements as "incoming". Match outgoing and incoming
-  elements and begin transition.
-- `transition.abandon()` - Same as MPA API.
+With this opt in, rather than the containers being siblings:
 
-This API will also capture elements with `page-transition-tag`, so it's possible
-for `createDocumentTransition` to be small and have CSS handle the rest:
+```
+::page-transition
+├─ …
+├─ ::page-transition-container(container)
+│  └─ ::page-transition-image-wrapper(container)
+│     └─ …
+└─ ::page-transition-container(child-item)
+   └─ ::page-transition-image-wrapper(child-item)
+      └─ …
+```
 
-```js
-document.createDocumentTransition(async (transition) => {
-  transition.start(async () => { await coolFramework.changeTheDOMToPageB() });
-});
+…the `child-item` would be nested in its closest parent that's also a transition element:
+
+```
+::page-transition
+├─ …
+└─ ::page-transition-container(container)
+   ├─ ::page-transition-image-wrapper(container)
+   │  └─ …
+   └─ ::page-transition-container(child-item)
+      └─ ::page-transition-image-wrapper(child-item)
+         └─ …
 ```
-Note that the `page-transition-tag` CSS property is consulted at the time
-`start` is called, as well as after the passed callback is finished running to
-determine the source and destination elements respectively.
 
-# Relation to `element()`
+## Better pseudo-element selectors
```

Comment:
Could we add the content vs full capture point also above this? It has come up as a common use-case when the shape of a UI widget is changing in a transition.